### ISSUE TITLE
Add container wait script for consul.

### ISSUE THREAD

**Issue by marioprudhomme** [DigitalState/Platform#70]
During the install, it seems sometimes some containers ups before consul / registrator is ready to accept incoming requests.

**Comment by Weglarz** [comment]
Hi Mario,
The same problem with containers records. I attached Log file after installation 

edit by admin: removed attached file

**Comment by Weglarz** [comment]
Please verify and give me answer

**Comment by Weglarz** [comment]
Hi Mario,
Still errors during installation process.
The results in attached log file.

edit by admin: removed attached file

**Comment by marioprudhomme** [comment]
What version of docker? We currently test with 18.06.1
What version of docker-compose? We currently test with 1.22.0
What version of operating system? We currently test with Windows 10 and
Ubuntu 18
What version of virtualization? On Windows 10, Pro version with HyperV is
required. On Ubuntu, just docker native works.

Also, please put inline your stacktrace or paste it on https://pastebin.com/ or github gists

**Comment by Weglarz** [comment]
Please fined attached file with the log of process installation:

docker version
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.4
 Git commit:        e68fc7a
 Built:             Fri Oct 19 19:43:14 2018
 OS/Arch:           linux/amd64

 docker-compose version
docker-compose version 1.23.1, build b02f1306
docker-py version: 3.5.0
CPython version: 3.6.7
OpenSSL version: OpenSSL 1.1.0f  25 May 2017

Still errors

edit by admin: removed attached file

**Comment by Weglarz** [comment]
Please support me

**Comment by Weglarz** [comment]
Please give me information what is status  of the issue  that  I described.

The installation procedure will be corrected?
I have to correct my installation environment?
Others ?

@marioprudhomme

**Comment by marioprudhomme** [comment]
It's fairely simple, please provide a stacktrace of your error logs via a github gist or paste it inline here, as per requested previously.

**Comment by Weglarz** [comment]
edit by admin: comments moved below for better readability

**Comment by StephenOTT** [comment]
reposting with proper markdown


```
Ubuntu 19.04

 

docker version

Client:

Version:           18.06.1-ce

API version:       1.38

Go version:        go1.10.1

Git commit:        e68fc7a

Built:             Fri Jan 25 14:33:51 2019

OS/Arch:           linux/amd64

Experimental:      false

 

Server:

Engine:

  Version:          18.06.1-ce

  API version:      1.38 (minimum version 1.12)

  Go version:       go1.10.4

  Git commit:       e68fc7a

  Built:            Thu Sep 27 02:39:50 2018

  OS/Arch:          linux/amd64

  Experimental:     false

 

docker-compose version

docker-compose version 1.23.1, build b02f1306

docker-py version: 3.5.0

CPython version: 3.6.7

OpenSSL version: OpenSSL 1.1.0f  25 May 2017

```

```
Stacktrace – last sentence:

TASK [Include compiled task list] ***********************************************************************************************************************************************************

included: /etc/ansible/task/app/proxy/container/up.yml for app => (item=/etc/ansible/task/app/proxy/container/up.yml)

included: /etc/ansible/task/app/discovery/container/up.yml for app => (item=/etc/ansible/task/app/discovery/container/up.yml)

included: /etc/ansible/task/app/formio/container/up.yml for app => (item=/etc/ansible/task/app/formio/container/up.yml)

included: /etc/ansible/task/app/camunda/container/up.yml for app => (item=/etc/ansible/task/app/camunda/container/up.yml)

included: /etc/ansible/task/app/services/container/up.yml for app => (item=/etc/ansible/task/app/services/container/up.yml)

included: /etc/ansible/task/app/forms/container/up.yml for app => (item=/etc/ansible/task/app/forms/container/up.yml)

included: /etc/ansible/task/app/cases/container/up.yml for app => (item=/etc/ansible/task/app/cases/container/up.yml)

included: /etc/ansible/task/app/assets/container/up.yml for app => (item=/etc/ansible/task/app/assets/container/up.yml)

included: /etc/ansible/task/app/identities/container/up.yml for app => (item=/etc/ansible/task/app/identities/container/up.yml)

included: /etc/ansible/task/app/records/container/up.yml for app => (item=/etc/ansible/task/app/records/container/up.yml)

included: /etc/ansible/task/app/authentication/container/up.yml for app => (item=/etc/ansible/task/app/authentication/container/up.yml)

included: /etc/ansible/task/app/cms/container/up.yml for app => (item=/etc/ansible/task/app/cms/container/up.yml)

 

TASK [Up proxy container] *******************************************************************************************************************************************************************

 

TASK [app/proxy/container/up : Up proxy container] ******************************************************************************************************************************************

changed: [app]

 

TASK [Up discovery container] ***************************************************************************************************************************************************************

 

TASK [app/discovery/container/up : Up discovery container] **********************************************************************************************************************************

changed: [app]

 

TASK [Retrieve proxy container ip] **********************************************************************************************************************************************************

included: /etc/ansible/task/app/proxy/container/property/ip.yml for app

 

TASK [Obtain proxy container ip] ************************************************************************************************************************************************************

changed: [app]

 

TASK [Up formio container] ******************************************************************************************************************************************************************

 

TASK [app/formio/container/up : Configure client host config] *******************************************************************************************************************************

changed: [app]

 

TASK [app/formio/container/up : Configure app host config] **********************************************************************************************************************************

changed: [app]

 

TASK [app/formio/container/up : Up formio container] ****************************************************************************************************************************************

changed: [app]

 

TASK [Retrieve proxy container ip] **********************************************************************************************************************************************************

included: /etc/ansible/task/app/proxy/container/property/ip.yml for app

 

TASK [Obtain proxy container ip] ************************************************************************************************************************************************************

changed: [app]

 

TASK [Up camunda container] *****************************************************************************************************************************************************************

 

TASK [app/camunda/container/up : Up camunda container] **************************************************************************************************************************************

changed: [app]

 

TASK [Retrieve proxy container ip] **********************************************************************************************************************************************************

included: /etc/ansible/task/app/proxy/container/property/ip.yml for app

 

TASK [Obtain proxy container ip] ************************************************************************************************************************************************************

changed: [app]

 

TASK [Up services container] ****************************************************************************************************************************************************************

 

TASK [app/services/container/up : Up services container] ************************************************************************************************************************************

changed: [app]

 

TASK [app/services/container/up : Waiting for services container to finalize installation] **************************************************************************************************

FAILED - RETRYING: Waiting for services container to finalize installation (60 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (59 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (58 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (57 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (56 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (55 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (54 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (53 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (52 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (51 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (50 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (49 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (48 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (47 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (46 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (45 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (44 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (43 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (42 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (41 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (40 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (39 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (38 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (37 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (36 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (35 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (34 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (33 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (32 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (31 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (30 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (29 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (28 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (27 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (26 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (25 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (24 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (23 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (22 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (21 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (20 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (19 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (18 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (17 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (16 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (15 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (14 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (13 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (12 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (11 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (10 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (9 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (8 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (7 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (6 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (5 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (4 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (3 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (2 retries left).

FAILED - RETRYING: Waiting for services container to finalize installation (1 retries left).

fatal: [app]: FAILED! => {"attempts": 60, "changed": false, "cmd": "docker-compose exec -T php pidof php-fpm | cat", "delta": "0:00:00.439817", "end": "2019-02-13 12:47:40.714486", "rc": 0, "start": "2019-02-13 12:47:40.274669", "stderr": "No container found for php_1", "stderr_lines": ["No container found for php_1"], "stdout": "", "stdout_lines": []}

 

PLAY RECAP **********************************************************************************************************************************************************************************

app                        : ok=131  changed=43   unreachable=0    failed=1

platform                   : ok=37   changed=7    unreachable=0    failed=0

 
```

**Comment by marioprudhomme** [comment]
@Weglarz 
If you want to install a DEV environment with the release 0.15.0 code, you will need to either:
1) Create a custom Ansible inventory (advanced) or 
2) Edit the already-made DEV Ansible inventory to make it point from the `develop` branch to the `release/0.15.0` branch (easy).

I suggest option 2.

Simply edit this line https://github.com/DigitalState/Platform/blob/release/0.15.0/platform/ansible/env/dev/vars/app/assets.yml#L5 , replacing `develop` with `release/0.15.0`

This needs to be done in that directory for each microservice files listed below:
```
assets.yml
authentication.yml
cases.yml
cms.yml
core.yml
discovery.yml
formio.yml
forms.yml
identities.yml
microservice.yml
proxy.yml
records.yml
services.yml
tasks.yml
tenants.yml
workflow.yml
```

The issue you are getting is due to using the Platform `0.15.0` branch with microservices in `develop` branches. Platform `develop` works with microservices in `develop` branches, Platform `0.15.0` works with microservices in `0.15.0` branch. 

Once the release is merged to `develop`, then you'll be able to use the DEV Ansible environment as-is without needing to modify versions. However, the DEV environment is mostly used by developers who is further developing the code.

Also note that you will need to clean your docker environment because currently, your server most likely cached the `develop` based docker images for all the microservices.

**Comment by marioprudhomme** [comment]
Closing this issue as the original intent of this issue has been solved 0ee094a3675eb51f43707113e6f0ba4ce101fb2e

@Weglarz 
If you are running into issues, please just open a single issue. It's easier to track the history.
